ofrak_core/ofrak/core/patch_maker/modifiers.py

```diff
@@ -115,23 +116,35 @@ async def modify(self, resource: Resource, config: PatchFromSourceModifierConfig
         target_linkable_bom_info = await target_program.make_linkable_bom(
             patch_maker,
             build_tmp_dir,
+            patch_bom.unresolved_symbols,
         )
+
         # To support additional dynamic references in user space executables
         # Create and use a modifier that will:
         # 1. Extend .got, add new entry
         # 2. Extend .got.plt, add new stub code
         # 3. If the DSO is not already listed in the load list for executable it must be extended and added.
         # 4. Provide the additional .got and .got.plt symbols to make_fem now that we have the locations
-        # NOTE: These external functions will probably be UND*
+        # NOTE: These external functions will probably be *UND*
         p = PatchRegionConfig(patch_bom.name + "_patch", patch_bom_segment_mapping)
         exec_path = os.path.join(build_tmp_dir, "output_exec")
-        fem = patch_maker.make_fem([(patch_bom, p), target_linkable_bom_info], exec_path)
+        fem = patch_maker.make_fem(
+            [(patch_bom, p), target_linkable_bom_info],
+            exec_path,
+        )
 
         await resource.run(
             SegmentInjectorModifier,
             SegmentInjectorModifierConfig.from_fem(fem),
         )
 
+        # Refresh LinkableBinary with the LinkableSymbols defined in this patch
+        target_binary = await resource.view_as(LinkableBinary)
```

Comment:
It seems like this should be grabbing the symbols from the final FEM. This assumes that all symbols (with definition in the object file itself) in all the BOMs are going to be injected as part of the FEM, which seems to be true, but I'm not certain.

This is more of a philosophical note - the FEM is what gets injected into the binary, so the FEM is what we should look at to see what symbol definitions are injected into the binary. This won't block this PR if left as-is.

Reply:
I agree with this comment. Doesn't seem to be a big lift to just take care of it here and now @marczalik ?

Reply:
Yes, I can make that change.